thundra/wrapper.go

```diff
@@ -51,8 +52,17 @@ func Wrap(handler interface{}, agent *thundra) interface{} {
 		go agent.catchTimeout(ctx, payload)
 
 		var args []reflect.Value
+
+		agent.executePreHooks(ctx, payload)
+
 		if takesContext {
-			args = append(args, reflect.ValueOf(ctx))
+			var ctxToPass context.Context
+			if plugin.CtxWithRootSpan != nil {
```

Comment:
There should not be trace specific logic here as this is common place. We need to find better approach.